agent/recorder.go

```diff
@@ -98,45 +106,38 @@ func (r *SpanRecorder) loop() error {
 					}
 				}
 				cTime = time.Now()
-				err, shouldExit := r.SendSpans()
+				err, shouldExit := r.sendSpans()
 				if err != nil {
 					r.logger.Printf("error sending spans: %v\n", err)
 				}
 				if shouldExit {
 					ticker.Stop()
 					r.t.Kill(err)
-					return nil
+					return err
```

Comment:
`r.t.Kill(err)` should stop the go routine, so no need to `return` here

Reply:
ok

Reply:
I have to return here because if I only Kill then the code behind the dying channel executes, and tries to send spans again.